### PR TITLE
Fix machine implementation updating

### PR DESCRIPTION
**Pull request by makhov**
Merge and patch existing machine objects to avoid owner conflict and/or admission webhook issues

### CODE REVIEW

File: internal/controller/controlplane/helper.go [hunk 1]
```diff
@@ -102,12 +103,45 @@ func (c *K0sController) createMachineFromTemplate(ctx context.Context, name stri
...
+	spec, _, _ := unstructured.NestedMap(existingMachineFromTemplate.Object, "spec")
+	patch := unstructured.Unstructured{Object: map[string]interface{}{
+		"spec": spec,
+	}}
+	data, err := patch.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	resourceType := strings.ToLower(machineFromTemplate.GetKind()) + "s"
```
Comment (apedriza) [inline internal/controller/controlplane/helper.go:137]: wdyt about find the apiresource.Name using resource kind and rely on the crd plural?

File: internal/controller/controlplane/k0s_controlplane_controller.go [hunk 2]
```diff
@@ -61,9 +62,10 @@ var ErrNewMachinesNotReady = fmt.Errorf("waiting for new machines")
 
 type K0sController struct {
 	client.Client
-	Scheme     *runtime.Scheme
-	ClientSet  *kubernetes.Clientset
-	RESTConfig *rest.Config
+	Scheme        *runtime.Scheme
+	ClientSet     *kubernetes.Clientset
+	DynamicClient dynamic.Interface
```
Comment (apedriza) [inline internal/controller/controlplane/k0s_controlplane_controller.go:67]: do we need `DynamicClient` ?

Comment (makhov) [reply internal/controller/controlplane/k0s_controlplane_controller.go]: Not anymore. Removed it, thanks!

File: internal/controller/controlplane/helper.go (hunk 1 above)
Comment (makhov) [reply internal/controller/controlplane/helper.go]: Changed it to more reliable way

**Review by apedriza — APPROVED** [review]

**Pull request merged**